toqito/state_props/is_abs_ppt.py

```diff
@@ -0,0 +1,155 @@
+                [eigvals[4] - eigvals[0], 2 * eigvals[2], eigvals[1] - eigvals[3]],
+                [eigvals[3] - eigvals[1], eigvals[1] - eigvals[3], 2 * eigvals[0]],
+            ]
+        )
+        L2 = np.array(
+            [
+                [2 * eigvals[5], eigvals[4] - eigvals[0], eigvals[2] - eigvals[1]],
+                [eigvals[4] - eigvals[0], 2 * eigvals[3], eigvals[1] - eigvals[2]],
+                [eigvals[2] - eigvals[1], eigvals[1] - eigvals[2], 2 * eigvals[0]],
+            ]
+        )
+        constraints.extend([L1, L2])
+    elif (d1, d2) == (3, 3):
+        # Two constraints: 3x3 matrices
+        # See QETLAB/AbsPPTConstraints and Hildebrand, Eq. (4)
+        # eigvals: lam1 >= ... >= lam9
+        L1 = np.array(
+            [
+                [2 * eigvals[8], eigvals[7] - eigvals[0], eigvals[6] - eigvals[1]],
+                [eigvals[7] - eigvals[0], 2 * eigvals[5], eigvals[4] - eigvals[2]],
+                [eigvals[6] - eigvals[1], eigvals[4] - eigvals[2], 2 * eigvals[3]],
+            ]
+        )
+        L2 = np.array(
+            [
+                [2 * eigvals[8], eigvals[7] - eigvals[0], eigvals[5] - eigvals[1]],
+                [eigvals[7] - eigvals[0], 2 * eigvals[6], eigvals[4] - eigvals[2]],
+                [eigvals[5] - eigvals[1], eigvals[4] - eigvals[2], 2 * eigvals[3]],
+            ]
+        )
+        constraints.extend([L1, L2])
+    else:
+        # Not implemented for higher dimensions
+        raise NotImplementedError("Absolutely PPT constraints for dimensions > 3x3 are not implemented.")
+    return constraints
+
+
+def is_abs_ppt(mat: np.ndarray, dim: None | int | list[int] = None, tol: float = 1e-8) -> bool | int:
+    r"""Determine whether a density matrix is absolutely PPT (PPT from spectrum).
+
+    :cite:`Hildebrand_2007_PPT, QETLAB_link`.
+
+    A density matrix is absolutely PPT if every unitary conjugation of it is PPT. This is a spectrum-based
+    property and can be checked using the Hildebrand/QETLAB constraints for small dimensions.
+
+    Examples
+    ==========
+    Consider the maximally mixed state in 2x2 dimensions:
+
+    .. jupyter-execute::
+
+        from toqito.states import max_mixed
+        from toqito.state_props import is_abs_ppt
+        rho = max_mixed(4)
+        is_abs_ppt(rho, [2, 2])
+
+    :param mat: The density matrix to check.
+    :param dim: The local dimensions (default: inferred as equal dims).
+    :param tol: Numerical tolerance for positive semidefinite checks.
+    :return: :code:`True` if absolutely PPT, :code:`False` if not, :code:`-1` if undecidable (for large dims).
+    :raises ValueError: If the input matrix is not square, not Hermitian, does not have trace 1, is not positive
+        semidefinite, or if the provided dimensions do not match the matrix size.
+
+    Note:
+        Returns -1 if the absolutely PPT property is undecidable for the given dimensions (i.e., for dimensions
+        other than 2x2, 2x3, 3x2, or 3x3).
+
```

Comment:
You need the references code block here such that the references are listed in the docs build.